Homework: Sprint 8

MaximusDenisov opened the pull request.

MaximusDenisov committed:
Layout completed for Main Screen and Settings Screen

MaximusDenisov committed:
Implemented button click handlers on Main Screen with mocks

MaximusDenisov committed:
Implemented screen navigation logic via button click listeners

Added Settings, Media, and Search screens with back button on Settings screen

MaximusDenisov committed:
Fixed string resources in @strings

i-masloed commented on .idea/.name:
Рекомендация: файлы директорий .idea и build стоит вынести в .gitignore

MaximusDenisov replied:
Исправил.

i-masloed commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -0,0 +1,105 @@
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="#3772E7"
```
Критично: цвет нужно вынести в ресурсы

MaximusDenisov replied:
Исправил.

i-masloed commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,172 @@
+    android:layout_height="match_parent"
+    android:orientation="vertical">
+
+    <LinearLayout
```
Эту комбинацию виджетов стоит заменить на Toolbar

MaximusDenisov replied:
@i-masloed спасибо большое за must have решения, в рамках курса этого не было, но это невероятно удобный и классный подход.
Спасибо за опыт!
Исправлено.

i-masloed commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,172 @@
+
+    </LinearLayout>
+
+    <LinearLayout
```
Эту комбинацию виджетов стоит заменить на один SwitchMaterial, растянуть его по ширине и добавить android:text="@string/dark_theme_setting"

MaximusDenisov replied:
@i-masloed , тоже очень крутая фишка, воспользовался и буду пользоваться в дальнейшем!
Исправлено.

i-masloed commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,172 @@
+            android:layout_marginEnd="@dimen/standart_indent_size" />
+    </LinearLayout>
+
+    <FrameLayout
```
эту комбинацию виджетов лучше заменить на один TextView, растянуть его на всю ширину и указать параметр drawableEnd для иконки. Аналогично с остальными на этом экране

MaximusDenisov replied:
My bad, исправил на TextView.
Исправлено.

MaximusDenisov committed:
Fixes based on feedback

Corrected .gitignore
Replaced old constructs with ToolBar and SwitchMaterial
Adjusted attributes in the settings list to use TextView
Moved color definitions to resources

MaximusDenisov committed:
Fixed Toolbar and applied it correctly

MaximusDenisov committed:
Fixed string for settings name
